src/useConstantCallback.js

```diff
@@ -0,0 +1,30 @@
+// @flow
+import * as React from 'react'
+
+/**
+ * Creates a callback, with dependencies, that will be
+ * instance === for the lifetime of the component.
+ */
+export default function useConstantCallback(callback, deps) {
+  // initialize refs on first render
+  const refs = deps.map(React.useRef)
```

Comment:
It might be useful to add this line as a safety net for sanity

```js
if (deps.length !== refs.length) {
  console.warn("Length of `deps` shouldn't change between renders")
}
```